ci/3-envs.yml

```diff
@@ -0,0 +1,427 @@
+    - put: pull-request
+      params:
+        path: pull-request
+        status: pending
+    - task: test-pull-request
+      file: pull-request/ci/tasks/unit.yml
+      input_mapping: { repo: pull-request }
+      on_success:
+        put: pull-request
+        params:
+          path: pull-request
+          status: success
+      on_failure:
+        put: pull-request
+        params:
+          path: pull-request
+          status: failure
+
+  - name: test-master
+    plan:
+      - get: master
+        resource: repo-master
+        trigger: true
+      - task: unit
+        file: master/ci/tasks/unit.yml
+        input_mapping: { repo: master }
+
+  - name: integrate-master
+    serial: true
+    plan:
+      - aggregate:
+        - get: master
+          resource: repo-master
+          trigger: true
+          passed: [ test-master ]
+      - task: integration-test
+        file: master/ci/tasks/integration.yml
+        input_mapping: {repo: master}
+
+  - name: version-master
+    serial: true
+    serial_groups: [ update-version ]
+    plan:
+      - aggregate:
+        - get: master
+          resource: repo-master
+          trigger: true
+          passed: [ integrate-master ]
+        - get: dev-version
+      - put: dev-version
+        params: { pre: dev }
+
+  - name: tag-master
+    serial: true
+    plan:
+      - aggregate:
+        - get: master
+          resource: repo-master
+        - get: dev-version
+          passed: [ version-master ]
+          trigger: true
+      - put: master
+        resource: repo-master
+        params:
+          repository: master
+          tag: dev-version/version
+
+  - name: ship-master
+    serial: true
+    plan:
+      - aggregate:
+        - get: master
+          resource: repo-master
+          trigger: true
+          passed: [ tag-master ]
+        - get: dev-version
+          passed: [ tag-master ]
+      - task: ship
+        file: master/ci/tasks/shipit.yml
+        input_mapping: { repo: master, version: dev-version }
+
+  - name: merge-master-to-pre-production
+    serial: true
+    plan:
+      - aggregate:
+        - get: master
+          resource: repo-master
+          passed: [ ship-master ]
+          # Uncomment to support automatic promotion
+          # from master to pre-production
+          # trigger: true
+        - get: pre-production
+          resource: repo-pre-production
+        - get: dev-version
+          params: { pre: rc }
+      - task: merge-pre-production-to-production
+        file: master/ci/tasks/merge-branch.yml
+        input_mapping: { from: master, to: pre-production }
+        output_mapping: { out: next-pre-production }
+        params:
+          GIT_EMAIL: {{git-email}}
+          GIT_NAME: {{git-name}}
+          NO_FF: true
+      - put: pre-production
+        resource: repo-pre-production
+        params:
+          repository: next-pre-production
+      - put: rc-version
+        params: { file: dev-version/version }
```

Comment:
if `rc-version` initialized with `x.y.z-rc.1`, tagged version will starts with `x.y.z-rc.2` because version-pre-production job always bumps `rc-version` with `{pre: rc}`.  To prevent this, we need to initialize `rc-version` with `x.y.z`.  However I would not like to store versions of `x.y.z` form to `rc-version` even if it was temporally.

what do you think??